sklearn/utils/tests/test_utils.py

```diff
@@ -317,7 +317,7 @@ def test_safe_indexing_2d_container_axis_1(array_type, indices_type, indices):
 
 @pytest.mark.parametrize("array_read_only", [True, False])
 @pytest.mark.parametrize("indices_read_only", [True, False])
-@pytest.mark.parametrize("array_type", ["array", "sparse", "dataframe"])
+@pytest.mark.parametrize("array_type", ["array", "sparse", "dataframe", "list"])
```

Comment:
The way ``_convert_container`` deals with lists is weird. When dtype is None, it returns list(container) which is fine for 1d ndarray but not for 2d array.
```py
X = np.ones((3, 3))
list(X)
# [array([1., 1., 1.]), array([1., 1., 1.]), array([1., 1., 1.])]
```

This part in _convert_container:
```py
    if constructor_name == "list":
        if dtype is None:
            return list(container)
        else:
            return np.asarray(container, dtype=dtype).tolist()
```
should be replaced by
```py
    if constructor_name == "list":
        return np.asarray(container, dtype=dtype).tolist()
```
I checked and it does what's expected in all the tests that uses it for list with dtype=None